src/Sentry/Internal/EnvironmentLocator.cs

```diff
@@ -1,17 +1,35 @@
 using System;
+using System.Diagnostics;
 
 namespace Sentry.Internal
 {
     internal static class EnvironmentLocator
     {
-        private static readonly Lazy<string?> Environment = new(Locate);
+        private static readonly Lazy<string?> FromEnvironmentVariable = new(LocateFromEnvironmentVariable);
 
         /// <summary>
         /// Attempts to locate the environment the app is running in.
         /// </summary>
         /// <returns>The Environment name or null, if it couldn't be located.</returns>
-        public static string? Current => Environment.Value;
+        public static string? Current => FromEnvironmentVariable.Value;
 
-        internal static string? Locate() => System.Environment.GetEnvironmentVariable(Constants.EnvironmentEnvironmentVariable);
+        internal static string? LocateFromEnvironmentVariable() =>
+            Environment.GetEnvironmentVariable(Constants.EnvironmentEnvironmentVariable);
+
+        public static string Resolve(SentryOptions options)
+        {
+            // Changing from `LocateFromEnvironmentVariable()` to `Current` fails tests?
```

Comment:
It sounds like the state of `Lazy<T>` persists across multiple tests because it's static.